### PR TITLE
doc: added some missing developer processes (#81066)

### PR DESCRIPTION
**Pull request by A-Herold**

### CODE REVIEW

File: doc/05_development_process.md [hunk 1]
```diff
@@ -221,6 +253,37 @@ PDF documents are stored within the shared file system of IOM. To get easy acces
...
+    # and create IOM again
+    devenv-cli.sh create iom
+
+## Apply Project Files
+
+Project files are any kind of files that have to be added to the IOM project image. More information about such files can be found in chapter *Project Files* of documentation of [Directory Structure of IOM Projects](https://github.com/intershop/iom-project-archetype/wiki/Directory-Structure-of-IOM-Projects#project-files).
+
+*devenv-4-iom* does not provide a process to roll out such files into a running IOM. Instead of it, you have to build the IOM projec image locally and to restart IOM within *devenv-4-iom* exactly the same way as described in previous chapter [Applying Custom Configurations](#apply-custom-configurations). The same preconditions apply for `IOM_IMAGE` and `IMAGE_PULL_POLICY` as already mentioned [above](#apply-custom-configurations).
+
+## Apply Test Data
```
Comment (mrc-mm) [inline doc/05_development_process.md:281]: I would rename this to "Import test articles"

**Review by mrc-mm — COMMENTED** [review]
Is there a good reason why "Apply SQL Scripts" does not work recursively?

**Comment by A-Herold** [comment]
> Is there a good reason why "Apply SQL Scripts" does not work recursively?

What do you mean exactly? What do you expect?

**Comment by mrc-mm** [comment]
I’d expected the script to process subdirectories recursively as with the handling of the stored prcedures.
This is not an issue, but not intuitiv.

**Comment by A-Herold** [comment]
Please have in mind, that there is also the command "apply sql-config". This command executes all SQL-scripts from the config directory recursively.
But I will add an issue for "apply sql-scripts" to work recursively, mainly to be able to easily update ALL stored procedures.

**Pull request merged**